src/App.tsx

```diff
@@ -1,24 +1,348 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
 import { UserWarning } from './UserWarning';
+import {
+  addTodo, deleteTodo, getTodos, updateTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { TodoItem } from './components/TodoItem';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { Error } from './components/Error';
+import { ErrMessage, FilterBy } from './types/Enums';
 
-const USER_ID = 0;
+const USER_ID = 11701;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [filter, setFilter] = useState(FilterBy.all);
+  const [errorWarning, setErrorWarning] = useState('');
+  const [pageIsLoaded, setPageIsLoaded] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isUpdatingId, setIsUpdatingId] = useState<number[]>([]);
+  const [isEditing, setIsEditing] = useState<number | null>(null);
+  const [editQuery, setEditQuery] = useState('');
+  const [editQueryPrev, setEditQueryPrev] = useState('');
+
+  const inputRef = useRef<HTMLBodyElement>(null);
+
+  const handleErrorSet = (errMessage: string) => {
+    setErrorWarning(errMessage);
+    setTimeout(() => {
+      setErrorWarning('');
+    }, 3000);
+  };
+
+  const closeError = () => {
+    setErrorWarning('');
+  };
+
+  const loadTodos = async () => {
+    try {
+      const loadedTodos = await getTodos(USER_ID);
+
+      setTodos(loadedTodos);
+      setPageIsLoaded(true);
+    } catch {
+      handleErrorSet(ErrMessage.loadTodo);
+    }
+  };
+
+  useEffect(() => {
+    loadTodos();
+  }, []);
+
+  useEffect(() => {
+    inputRef.current?.focus();
+  }, [inputRef, pageIsLoaded, todos]);
+
+  const visibleTodos = (filterBy: FilterBy) => {
+    if (todos) {
+      switch (filterBy) {
+        case FilterBy.all:
+          return todos;
+        case FilterBy.active:
+          return todos.filter(todo => (
+            !todo.completed
+          ));
+        case FilterBy.completed:
+          return todos.filter(todo => (
+            todo.completed
+          ));
+        default: return todos;
+      }
+    }
+
+    return todos;
+  };
+
+  const handleFormSubmit = async (e?: Event) => {
+    e?.preventDefault();
+    if (!query.trim().length) {
+      handleErrorSet(ErrMessage.titleEmpty);
+
+      return;
+    }
+
+    const newTodo = {
+      id: todos.length + 10 || 10,
+      userId: USER_ID,
+      title: query.trim(),
+      completed: false,
+    };
+
+    setTempTodo({ ...newTodo, id: 0 });
+
+    try {
+      setPageIsLoaded(false);
+      await addTodo('/todos', newTodo);
```

Comment:
You don't need to pass `id` if you create a new todo because it wouldn't work.
You should create a new todo only with the next fields:
- userId
- title
- completed

So, it's better to have two separate handlers for creating and updating.
For example, `handleCreateTodo`, `handleUpdateTodo`.
Or you have to implement additional logic here to divide if user wants to create or update Todo.

Reply:
i am confused , how can i not make id for new todo ?

Reply:
check [here](https://mate-academy.github.io/fe-students-api/todos.html) what data you should send to server for creating a new todo